### PR TITLE
Add system clock and timezone API.

### PR DESCRIPTION
**Pull request by cabelitos**
Depends on https://github.com/solettaproject/soletta/pull/1141

Please, do not review the commit 8f27cf8 here.

### CODE REVIEW

File: src/lib/common/sol-platform-linux-common.c [hunk 1]
```diff
@@ -776,3 +798,99 @@ sol_platform_linux_uevent_unsubscribe(const char *action, const char *subsystem,
...
+{
+    return (int64_t)time(NULL);
+}
+
+static bool
+system_clock_changed(void *data, int fd, uint32_t active_flags)
+{
+    char buf[4096];
+
+    if (read(timer_ctx.fd, buf, sizeof(buf)) >= 0)
```
Comment (lpereira) [inline src/lib/common/sol-platform-linux-common.c]: If an error such as `EINTR` is received, the fd will be closed, is this by design?

File: src/lib/common/sol-platform-linux-common.c [hunk 2]
```diff
@@ -776,3 +798,99 @@ sol_platform_linux_uevent_unsubscribe(const char *action, const char *subsystem,
...
+    struct itimerspec spec;
+
+    if (timer_ctx.watcher)
+        return 0;
+
+    timer_ctx.fd = timerfd_create(CLOCK_REALTIME, TFD_CLOEXEC);
+    SOL_INT_CHECK(timer_ctx.fd, < 0, -errno);
+
+    memset(&spec, 0, sizeof(struct itimerspec));
+    spec.it_value.tv_sec += 0xfffffff0;
```
Comment (lpereira) [inline src/lib/common/sol-platform-linux-common.c]: Why the magic value? Might be nice to have a comment explaining what the hack does.

File: src/lib/common/sol-platform-linux-common.c [hunk 3]
```diff
@@ -776,3 +798,99 @@ sol_platform_linux_uevent_unsubscribe(const char *action, const char *subsystem,
...
+        SOL_WRN("Could not readlink /etc/localtime");
+        return NULL;
+    }
+    timezone_str[n] = '\0';
+    if (strstartswith(timezone_str, "../usr/share/zoneinfo/"))
+        offset = strlen("../usr/share/zoneinfo/");
+    else if (strstartswith(timezone_str, "/usr/share/zoneinfo/"))
+        offset = strlen("/usr/share/zoneinfo/");
+    else {
+        SOL_WRN("The timzeon is not a link to /usr/share/zoneinfo/");
```
Comment (lpereira) [inline src/lib/common/sol-platform-linux-common.c]: s/timzeon/timezone/

File: src/lib/common/sol-platform-linux-common.c (hunk 1 above)
Comment (cabelitos) [inline src/lib/common/sol-platform-linux-common.c]: This is by design. If the execution reach this code it means that the system clock changed and I need to register a new one.

File: src/lib/common/sol-platform-linux-common.c (hunk 3 above)
Comment (cabelitos) [inline src/lib/common/sol-platform-linux-common.c]: K

File: src/lib/common/sol-platform-linux-common.c (hunk 2 above)
Comment (cabelitos) [inline src/lib/common/sol-platform-linux-common.c]: Sorry, I will explain.

**Comment by cabelitos** [comment]
Closing in favor of #1164

**Pull request closed without merging**